build_runner/lib/src/generate/build_impl.dart

```diff
@@ -192,7 +197,25 @@ class _SingleBuild {
         _resolvers = buildImpl._resolvers,
         _resourceManager = buildImpl._resourceManager,
         _verbose = buildImpl._verbose,
-        _writer = buildImpl._writer;
+        _writer = buildImpl._writer {
+    hungActionsHeartbeat = new HungActionsHeartbeat(() {
```

Comment:
[optional] I wonder if we can move the details to another file? I see its using variables from this file so maybe it would be too much overhaul for this PR but we should consider whether we can do more to separate out the behavior of the build from the UX of the logging...

Reply:
I get the sentiment but I don't think its really worth doing in this case - this is just significantly simpler than the other options.

Reply:
I think long term we might want to consider pulling out all of the UX focused code from this file.